browser/BUILD.gn

```diff
@@ -284,7 +268,37 @@ source_set("browser_process") {
   }
 
   if (brave_rewards_enabled) {
+    sources += [
+      "brave_rewards/rewards_tab_helper.cc",
+      "brave_rewards/rewards_tab_helper.h",
+      "brave_rewards/rewards_service_factory.cc",
+      "brave_rewards/rewards_service_factory.h",
+    ]
+
+    if (enable_extensions) {
+      sources += [
+        "brave_rewards/extension_rewards_service_observer.cc",
+        "brave_rewards/extension_rewards_service_observer.h",
+        "brave_rewards/extension_rewards_notification_service_observer.cc",
+        "brave_rewards/extension_rewards_notification_service_observer.h",
+      ]
+    }
+
+    if (is_android) {
+      sources += [
+        "brave_rewards/android/brave_rewards_native_worker.cc",
+        "brave_rewards/android/brave_rewards_native_worker.h",
+      ]
+    } else {
+      sources += [
+        "brave_rewards/tip_dialog.cc",
+        "brave_rewards/tip_dialog.h",
+      ]
+    }
+
     deps += [
+      "//brave/components/brave_rewards/browser",
+      "//brave/vendor/bat-native-ledger",
```

Comment:
can you limit this to only a target for the mojo structs and also add visibility settings to the main target so it can only be included in components/brave_rewards? Also I think this is only needed in `is_android`

Reply:
done